### PR TITLE
NO-JIRA: update library-go

### PR DESCRIPTION
**Pull request by tjungblu**
for openshift/library-go#1743

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,23 +14,23 @@ require (
...
-	k8s.io/client-go v0.30.0
-	k8s.io/component-base v0.30.0
+	k8s.io/api v0.30.1
+	k8s.io/apiextensions-apiserver v0.30.1
+	k8s.io/apimachinery v0.30.1
+	k8s.io/apiserver v0.30.1
+	k8s.io/client-go v0.30.1
+	k8s.io/component-base v0.30.1
 	k8s.io/klog/v2 v2.120.1
 	k8s.io/pod-security-admission v0.30.0
```
Comment (p0lyn0mial) [inline go.mod:35]: could you bump it to 0.30.1 as well ?

Comment (tjungblu) [reply go.mod:35]: absolutely can

**Review by vrutkovs — APPROVED** [review]
/approve
/lgtm

**Comment by p0lyn0mial** [comment]
@tjungblu, CI fails on:
```
go: inconsistent vendoring in /go/src/github.com/openshift/cluster-kube-apiserver-operator:
```

it wants

```
To sync the vendor directory, run:
go mod vendor
```

**Comment by tjungblu** [comment]
fixed, sorry, forgot to run the commands again

**Pull request merged**